src/option_parser.cr

```diff
@@ -417,6 +425,15 @@ class OptionParser
         @invalid_option.call(arg)
       end
     end
+  ensure
+    @flags = old_flags.not_nil!
+    @handlers = old_handlers.not_nil!
+    @stop = false
+    @banner = old_banner
+    @unknown_args = old_unknown_args
+    @missing_option = old_missiong_option.not_nil!
+    @invalid_option = old_invalid_option.not_nil!
+    @before_each = old_before_each
```

Comment:
Wouldn't be better to use `begin ... end` block instead of using `.not_nil!`?

Reply:
I don't think it's worth the extra indent. They're semantically equivalent anyway.

Reply:
OK then how about clumping it all into one variable (tuple? record? maybe store the whole state in a sub-record in the first place?)

Reply:
It should be either as is or without the reset.

Reply:
@RX14 Using `.not_nil!` should be treated as a last resort, so if there's a way to avoid it, even at the "cost" of an extra indent, then I'd say it's worth it.

Reply:
Just put the begin after you save all those variables. I agree with Sija.